test/boost/database_test.cc

```diff
@@ -1127,7 +1127,7 @@ SEASTAR_THREAD_TEST_CASE(max_result_size_for_unlimited_query_selection_test) {
         database_test tdb(db);
         for (const auto& [sched_group, expected_max_size] : scheduling_group_and_expected_max_result_size) {
             with_scheduling_group(sched_group, [&db, sched_group = sched_group, expected_max_size = expected_max_size] {
-                const auto max_size = db.get_unlimited_query_max_result_size();
+                const auto max_size = db.get_query_max_result_size();
```

Comment:
the `operator==` for `max_result_size` compares all member variables of this struct. that's why this test now fails because all the return values returned by `database::get_query_max_result_size()` have non-zero `page_size` -- they are either configured with `query_page_size_in_bytes` or hardwired to `maximum_result_size`. so, probably we could initialize the expected page size also when creating `user_max_result_size` and `system_max_result_size`?